index.js

```diff
@@ -4,15 +4,30 @@
 module.exports = {
   name: 'ember-hbs-minifier',
 
-  setupPreprocessorRegistry(type, registry) {
-    if (type === 'parent') {
-      let HbsMinifierPlugin = require('./hbs-minifier-plugin');
+  _getMinifierOptions() {
+    return (this.parent && this.parent.options) || (this.app && this.app.options) || {};
+  },
+
+  _setupPreprocessorRegistry(app) {
+    let registry = app.registry;
+    let options = this._getMinifierOptions();
+    let config = options['ember-hbs-minifier'] || {};
 
-      registry.add('htmlbars-ast-plugin', {
-        name: 'hbs-minifier-plugin',
-        plugin: HbsMinifierPlugin,
-        baseDir() { return __dirname; }
-      });
-    }
+    let HbsMinifierPlugin = require('./hbs-minifier-plugin')(config.skip || {});
+    registry.add('htmlbars-ast-plugin', {
+      name: 'hbs-minifier-plugin',
+      plugin: HbsMinifierPlugin,
+      baseDir() { return __dirname; },
+      cacheKey() { return 'ember-hbs-minifier'; }
```

Comment:
as I mentioned before, I think this needs to be computed based on the `config` hash, otherwise config changes won't result in cache invalidations

Reply:
Shall we generate a [MD5-hash](https://github.com/puleos/object-hash#hashmd5value) based on the `config`?